tests/periph_gpio/main.c

```diff
@@ -203,6 +207,97 @@ static int toggle(int argc, char **argv)
     return 0;
 }
 
+static void bench_print_time(uint32_t time, unsigned long runs, const char *name)
+{
+    uint32_t full = (time / runs);
+    uint32_t div  = (time - (full * runs)) / (runs / 1000);
+
+    printf("%11s: %9luus  ---  %2u.%03uus per call\n", name,
+           (unsigned long) time, (unsigned)full, (unsigned)div);
+}
+
+static int bench(int argc, char **argv)
+{
+    if (argc < 3) {
+        printf("usage: %s <port> <pin>\n", argv[0]);
+        return 1;
+    }
+
+    uint32_t time;
+    gpio_t pin = GPIO_PIN(atoi(argv[1]), atoi(argv[2]));
+    unsigned long runs = BENCH_RUNS_DEFAULT;
+    if (argc > 3) {
+        runs = (unsigned long)atoi(argv[3]);
+    }
+
+    puts("\nGPIO driver run-time performance benchmark\n");
+    /* just a plain NOP loop (for reference) */
+    unsigned state = irq_disable();
+    time = xtimer_now_usec();
+    for (unsigned long i = 0; i < runs; i++) {
+        __asm("nop");
+    }
+    time = (xtimer_now_usec() - time);
+    irq_restore(state);
+    bench_print_time(time, runs, "nop loop");
+
+    /* gpio_set */
+    state = irq_disable();
+    time = xtimer_now_usec();
+    for (unsigned long i = 0; i < runs; i++) {
+        gpio_set(pin);
+    }
+    time = (xtimer_now_usec() - time);
+    irq_restore(state);
+    bench_print_time(time, runs, "gpio_set");
+
+    /* gpio_clear */
+    state = irq_disable();
+    time = xtimer_now_usec();
+    for (unsigned long i = 0; i < runs; i++) {
+        gpio_clear(pin);
+    }
+    time = (xtimer_now_usec() - time);
+    irq_restore(state);
+    bench_print_time(time, runs, "gpio_clear");
+
+    /* gpio_clear */
+    state = irq_disable();
+    time = xtimer_now_usec();
+    for (unsigned long i = 0; i < runs; i++) {
+        gpio_toggle(pin);
+    }
+    time = (xtimer_now_usec() - time);
+    irq_restore(state);
+    bench_print_time(time, runs, "gpio_toggle");
+
+    /* gpio_clear */
+    state = irq_disable();
+    time = xtimer_now_usec();
+    for (unsigned long i = 0; i < runs; i++) {
+        gpio_write(pin, 1);
+    }
+    time = (xtimer_now_usec() - time);
+    irq_restore(state);
+    bench_print_time(time, runs, "gpio_write");
+
+    /* gpio_read */
+    state = irq_disable();
+    time = xtimer_now_usec();
+    for (unsigned long i = 0; i < runs; i++) {
+        int tmp = gpio_read(pin);
```

Comment:
why save the return and then void it? does it prevent optimizing the function call away?